src/Promitor.Core.Scraping/Scraper.cs

```diff
@@ -103,10 +103,60 @@ public async Task ScrapeAsync(ScrapeDefinition<IAzureResourceDefinition> scrapeD
             }
         }
 
+        public async Task BatchScrapeAsync(BatchScrapeDefinition<IAzureResourceDefinition> batchScrapeDefinition)
+        {  
+            // would the large volume of JSON be an issue? Can it be handled by the SDK? 
+            if (batchScrapeDefinition == null)
+            {
+                throw new ArgumentNullException(nameof(batchScrapeDefinition));
+            }
+
+            var aggregationInterval = batchScrapeDefinition.ScrapeDefinitionBatchProperties.AggregationInterval;
+            if (aggregationInterval == null)
+            {
+                throw new ArgumentNullException(nameof(scrapeDefinition));
+            }
+
+            try
+            {
+                var aggregationType = batchScrapeDefinition.ScrapeDefinitionBatchProperties.AzureMetricConfiguration.Aggregation.Type;
+                var scrapeDefinitions = batchScrapeDefinition.ScrapeDefinitions;
+                var scrapedMetricResult = await BatchScrapeResourceAsync(
+                    batchScrapeDefinition.ScrapeDefinitionBatchProperties.SubscriptionId,
+                    batchScrapeDefinition,
+                    aggregationType,
+                    aggregationInterval.Value);
+                
+                foreach (int i in Enumerable.Range(0, scrapedMetricResult.Count))
+                {
+                    var scrapedMetricResult = scrapedMetricResult[i];
+                    var scrapeDefinition = scrapeDefinitions[i];
+                    LogMeasuredMetrics(scrapeDefinition, scrapedMetricResult, aggregationInterval);
+
+                    await _metricSinkWriter.ReportMetricAsync(scrapeDefinition.PrometheusMetricDefinition.Name, scrapeDefinition.PrometheusMetricDefinition.Description, scrapedMetricResult);
+
+                    await ReportScrapingOutcomeAsync(scrapeDefinition, isSuccessful: true, isBatchJob: true) ;
+                }

```

Comment:
We may need to fan this out to running tasks in parallel for every result, but can do that in a later update

Reply:
Yes agreed, there are potentially a lot of results to stuff into a single thread. While we are on this topic, I believe timing metric(s) for task execution would be great to add. You know, so we can measure outcomes quantitively